lms/views/api/vitalsource.py

```diff
@@ -1,32 +1,39 @@
+import re
+
+from pyramid.httpexceptions import HTTPClientError
 from pyramid.view import view_config, view_defaults
 
 from lms.security import Permissions
-from lms.services.exceptions import ProxyAPIError
-from lms.views.api.vitalsource_sample_data import book_data, toc_data
 
 
 @view_defaults(renderer="json", permission=Permissions.API)
 class VitalSourceAPIViews:
     def __init__(self, request):
         self.request = request
+        self.vitalsource_service = request.find_service(name="vitalsource")
 
     @view_config(route_name="vitalsource_api.books.info")
     def book_info(self):
         book_id = self.request.matchdict["book_id"]
+        if not self._is_valid_book_id(book_id):
+            raise HTTPClientError("Invalid `book_id`. It must only contain [0-9A-Z-].")
 
-        # In future this will fetch data from VitalSource.
-        if book_id not in book_data:
-            raise ProxyAPIError(f"Book {book_id} not found")
-
-        return book_data[book_id]
+        book_info = self.vitalsource_service.book_info(book_id)
+        return {
+            "id": book_info["vbid"],
+            "title": book_info["title"],
+            "cover_image": book_info["resource_links"]["cover_image"],
+        }
 
     @view_config(route_name="vitalsource_api.books.toc")
     def table_of_contents(self):
         book_id = self.request.matchdict["book_id"]
+        if not self._is_valid_book_id(book_id):
```

Comment:
Perhaps worth extracting the logic for extract the `book_id` parameter, validating it and throwing an error if invalid into a single helper, to ensure consistency across different endpoints.

Reply:
Yes, good point, added a helper function to do the extracting (and posible raising) but kept the second function that just does the regexp.